From 52870e8a264653d7921e8a4edff56d4b050614d8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 27 Jan 2018 13:11:36 +0100
Subject: [PATCH] Determine architecture via 'uname -m'

'uname -p' only gives unknown on x86_64, i386, arm6l (rpi) and aarch64
(scaleways).
---
 virt-what.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/virt-what.in b/virt-what.in
index 29b7b0d..2011ff4 100644
--- a/virt-what.in
+++ b/virt-what.in
@@ -102,9 +102,9 @@ cpuid=$(virt-what-cpuid-helper)
 dmi=$(LANG=C dmidecode 2>&1)
 
 # Architecture.
-# Note for the purpose of testing, we only call uname with -p option.
+# Note for the purpose of testing, we only call uname with -m option.
 
-arch=$(uname -p | sed -e 's/i.86/i386/' | sed -e 's/arm.*/arm/')
+arch=$(uname -m | sed -e 's/i.86/i386/' | sed -e 's/arm.*/arm/')
 
 # Check for VMware.
 # cpuid check added by Chetan Loke.
-- 
2.16.1

